x-pack/plugins/alerting/common/rule.ts

```diff
@@ -18,6 +18,9 @@ import { RuleSnooze } from './rule_snooze_type';
 export type RuleTypeState = Record<string, unknown>;
 export type RuleTypeParams = Record<string, unknown>;
 
+// rule type defined alert fields to persist in alerts index
+export type RuleAlertData = Record<string, unknown>;
```

Comment:
Added a new generic for rule types to specify their rule type specific alert schema. This probably isn't strictly needed until we implement the followup PR for https://github.com/elastic/kibana/issues/156443